spec/inertia/conditional_sharing_spec.rb

```diff
@@ -0,0 +1,27 @@
+RSpec.describe "conditionally shared data in a controller", type: :request do
+  it 'does not leak data between requests' do
+    get conditional_share_index_path, headers: {'X-Inertia' => true}
+    expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
+      index_only_prop: 1,
+      normal_shared_prop: 1,
+    })
+
+    # NOTE: we actually have to run the show action twice since the new implementation
+    # sets up a before_action within a before_action to share the data.
+    # In effect, that means that the shared data isn't rendered until the second time the action is run.
+    get conditional_share_show_path, headers: {'X-Inertia' => true}
+    get conditional_share_show_path, headers: {'X-Inertia' => true}
+    expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
+      normal_shared_prop: 1,
+      show_only_prop: 1,
+      conditionally_shared_show_prop: 1,
+    })
+
+    get conditional_share_index_path, headers: {'X-Inertia' => true}
+    expect(JSON.parse(response.body)['props'].deep_symbolize_keys).to eq({
+      index_only_prop: 1,
+      normal_shared_prop: 1,
+      conditionally_shared_show_prop: 1, # TODO: Because we call before_action twice, this is leaking.
```

Comment:
@bknoles i have cherry picked this failling spec, for now i add a TODO untill we think about

Reply:
In this one i think the best solution is already implement one of this approach to run the inertia_share to only some controller method https://github.com/inertiajs/inertia-rails/issues/115#issuecomment-2081728942.

`before_actions_options` is a good explanatory name, but i think maybe is too big i would go for only `options` or `share_options` to make less conflict  with existing code

Reply:
Good ideia i will keep with the test